### PR TITLE
perf: declare a count for tranches length

### PR DESCRIPTION
**Pull request by andreivladbrg**
Depends on https://github.com/sablier-labs/v2-periphery/pull/357

### Changelog

Declare a `count` variable to optimize loading it into memory on every iteration. It was necessary to add a new scope block to prevent the stack too deep error.

### CODE REVIEW

**Review by smol-ninja — APPROVED** [review]
LGTM except the build failed.

**Review by PaulRBerg — APPROVED** [review]
good, will merge later once the other PR is merged in

**Pull request merged**
